apps/matrix/main.py

```diff
@@ -148,16 +154,26 @@ def display_rooms(self):
 			# Right arrow 	-> Write message to room
 			menu_contents.append([
 				room_name,
-				lambda x=current_room: self.display_messages(x),
-				lambda x=current_room: self.write_message(x)
+				lambda r=current_room: self.display_messages(r),
+				lambda r=current_room: self.display_room_members(r)
 			])
 
 		menu_contents.append(["Settings", self.show_settings])
 		Menu(menu_contents, self.i, self.o, name="Matrix app main menu").activate()
 
+	# Display a menu of settings for the app
 	def show_settings(self):
-		mc = [["Log out", self.logout]]
-		Menu(mc, self.i, self.o, catch_exit=False, name="Matrix app settings menu").activate()
+		mc = [
+				["Log out", self.logout],
+				["Hide join/leave messages", lambda: self._change_setting("show_join_leave_messages", False)],
```

Comment:
Please, re-do this to toggle the setting instead (use [`contents_hook`](https://zpui.readthedocs.io/en/devel/howto.html#auto-regenerating-menu-contents) for autogenerating the correct label), so that there's only one entry instead of two - and it also shows the status